src/dbc.rs

```diff
@@ -186,142 +403,120 @@ mod tests {
     #[allow(clippy::too_many_arguments)]
     #[quickcheck]
     fn prop_dbc_validation(
-        n_inputs: NonZeroTinyInt,      // # of input DBC's
-        n_valid_sigs: TinyInt,         // # of valid sigs
-        n_wrong_signer_sigs: TinyInt,  // # of valid sigs from unrecognized authority
-        n_wrong_msg_sigs: TinyInt,     // # of sigs from recognized authority signing wrong message
-        n_extra_input_sigs: TinyInt,   // # of sigs for inputs not part of the transaction
-        extra_output_amount: TinyInt,  // Artifically increase output dbc value
-        n_add_random_parents: TinyInt, // # of random parents to add to output DBC
-        n_drop_parents: TinyInt,       // # of valid parents to drop from output DBC
+        n_inputs: NonZeroTinyInt,     // # of input DBC's
+        n_valid_sigs: TinyInt,        // # of valid sigs
+        n_wrong_signer_sigs: TinyInt, // # of valid sigs from unrecognized authority
+        n_wrong_msg_sigs: TinyInt,    // # of sigs from recognized authority signing wrong message
+        n_extra_input_sigs: TinyInt,  // # of sigs for inputs not part of the transaction
+        extra_output_amount: TinyInt, // Artifically increase output dbc value
     ) -> Result<(), Error> {
-        let amount = 100;
-        let genesis_owner = crate::bls_dkg_id();
-        let genesis_key = genesis_owner.public_key_set.public_key();
-
-        let key_manager = SimpleKeyManager::new(
-            SimpleSigner::from(genesis_owner.clone()),
-            genesis_owner.public_key_set.public_key(),
-        );
-        let mut genesis_node = MintNode::new(key_manager);
+        let mut rng8 = rand8::rngs::StdRng::from_seed([0u8; 32]);
+        let mut rng = rand::rngs::StdRng::from_seed([0u8; 32]);
 
-        let genesis = genesis_node.issue_genesis_dbc(amount).unwrap();
+        let amount = 100;
 
-        let genesis_sig = genesis
-            .public_key_set
-            .combine_signatures(vec![genesis.transaction_sig.threshold_crypto()])
-            .unwrap();
+        let (mint_node, mut spentbook, genesis, _genesis_dbc) =
+            GenesisBuilderMock::init_genesis_single(amount, &mut rng, &mut rng8)?;
 
-        let genesis_dbc = Dbc {
-            content: genesis.dbc_content,
-            transaction: genesis.transaction,
-            transaction_sigs: BTreeMap::from_iter([(
-                crate::genesis_dbc_input(),
-                (genesis_key, genesis_sig),
-            )]),
-        };
+        let input_owners: Vec<OwnerOnce> = (0..=n_inputs.coerce())
```

Comment:
Did you mean n_inputs + 1 owners?
```suggestion
        let input_owners: Vec<OwnerOnce> = (0..n_inputs.coerce())
```

Reply:
yeah, that works.